### DISCUSSION TITLE
Unable to pull data from API, Access Token Invalid

### DISCUSSION THREAD

**Discussion by bhuv1p** [ercot/api-specs#42] [Q&A]
Hi,

I have signed up on the ERCOT website and got an API key. Whey I try to pull data from any end point using a simple get request in python, I get a 401 error saying my access key is invalid. I have tried using both primary and secondary keys but nothing seems to work. Can I know how to solve this issue?

```
import requests

url = 'https://api.ercot.com/api/public-reports/np4-738-cd/spp_actual_5min_avg_values'
headers = {
    'Ocp-Apim-Subscription-Key': '' # key here
}

response = requests.get(url, headers=headers)
```

A sample query would also be appreciated. Thanks in advance.

Best,
Bhuv.

**Comment by mackermann-ercot** [comment]
Hi Bhuv,

Thanks for posting your question.  There was a similar question by psatVol [here](https://github.com/ercot/api-specs/discussions/34) which I think you will find quite useful. 

Regards,
mackermann, ERCOT
